bitcoin/src/blockdata/transaction.rs

```diff
@@ -497,24 +512,29 @@ impl TxOut {
     pub const NULL: Self =
         TxOut { value: Amount::from_sat(0xffffffffffffffff), script_pubkey: ScriptBuf::new() };
 
-    /// The weight of the txout in witness units
+    /// The weight of this output.
     ///
-    /// Keep in mind that when adding a TxOut to a transaction, the total weight of the transaction
-    /// might increase more than `TxOut::weight`. This happens when the new output added causes
-    /// the output length `VarInt` to increase its encoding length.
+    /// Keep in mind that when adding a [`TxOut`] to a [`Transaction`] the total weight of the
+    /// transaction might increase more than `TxOut::weight`. This happens when the new output added
+    /// causes the output length `VarInt` to increase its encoding length.
     pub fn weight(&self) -> Weight {
-        let script_len = self.script_pubkey.len();
-        // In vbytes:
-        // value (8) + script varint len + script push
-        Weight::from_non_witness_data_size((8 + VarInt::from(script_len).len() + script_len) as u64)
+        // FIXME: Is this vbyte/byte comment correct, sane, and meaningful?
+        // For outputs size is equivalent to virtual size since it is the same for legacy and segwit.
+        // FIXME: Is this unchecked usage correct?
+        Weight::from_vb_unchecked(self.size() as u64) // Unchecked because size() cannot overflow.
```

Comment:
The comment would be simpler as `size is equivalent to virtual size since all bytes of a TxOut are non-witness bytes`.

As for the claims of overflow, strictly speaking no. When convirting from vsize to weight there is a *4 multiplier, which could cause an overflow even if `size` does not overflow.

IMO we shoud panic here and document that if the script size exceeds 2^62 then it'll panic.

Reply:
Cool thanks, will work in the suggestions.

Reply:
Would it not be better to have all the `weight` functions return `Option<Weight>` and add `weight_unchecked` functions?

Reply:
I dunno, this feels like a crappy API given that under normal usage weights will never come within a factor of a million of overflowing.

We could have `weight` functions that panic or truncate and `weight_checked` ones that don't.

Reply:
I solved this by just using saturating add/mull and commenting that its for defensive reasons.

Reply:
I think panicking would be better because if someone hits it it means something is very wrong somewhere else. Finding why the value is `u64::MAX` is much harder.